### PR TITLE
Add name_query filter to ListLabels

### PR DESCRIPTION
**Pull request by nicksnyder**
This enables users to find matching labels if they only type in a substring of the label name (this is for the label selector in a module).

### CODE REVIEW

File: buf/registry/module/v1/label_service.proto [hunk 1]
```diff
@@ -115,6 +115,8 @@ message ListLabelsRequest {
   //
   // TODO: Add custom CEL validation to validate the status field is one of DISABLED, PASSED, APPROVED.
   repeated CommitCheckStatus commit_check_statuses = 5 [(buf.validate.field).repeated.items.enum.defined_only = true];
+  // Only return Labels with a name that contains this string.
+  string name = 6 [(buf.validate.field).string.max_len = 250];
```
Comment (nicksnyder) [inline buf/registry/module/v1/label_service.proto:119]: Could also make this more generically named `query`.

Comment (mfridman) [reply buf/registry/module/v1/label_service.proto:119]: Would it make sense to have a `ListLabelsFilter` message with all the possible filters for a given request?

```proto
ListLabelsFilter filters = 6;
```

**Review by bufdev — APPROVED** [review]

File: buf/registry/module/v1/label_service.proto (hunk 1 above)
Comment (nicksnyder) [reply buf/registry/module/v1/label_service.proto]: In a live convo, decided that we wanted to keep the datastructure flat and instead to call it `name_query`.

File: buf/registry/module/v1/label_service.proto [hunk 2]
```diff
@@ -115,6 +115,8 @@ message ListLabelsRequest {
   //
   // TODO: Add custom CEL validation to validate the status field is one of DISABLED, PASSED, APPROVED.
   repeated CommitCheckStatus commit_check_statuses = 5 [(buf.validate.field).repeated.items.enum.defined_only = true];
+  // Only return Labels with a name that contains this string.
+  string name_query = 6 [(buf.validate.field).string.max_len = 250];
```
Comment (saquibmian) [inline buf/registry/module/v1/label_service.proto:119]: Is this match case?

**Pull request merged**